google-cloud-storage/src/main/java/com/google/cloud/storage/ParallelCompositeUploadBlobWriteSessionConfig.java

```diff
@@ -628,6 +660,71 @@ protected String fmtFields(String randomKey, String nameDigest, String partRange
     }
   }
 
+  /**
+   * @see #withPartMetadataFieldDecorator(PartMetadataFieldDecorator)
+   * @since 2.35.1 This new api is in preview and is subject to breaking changes.
+   */
+  @BetaApi
+  @Immutable
+  public abstract static class PartMetadataFieldDecorator implements Serializable {
+
+    abstract PartMetadataFieldDecoratorInstance newInstance(Clock clock);
+
+    /**
+     * A strategy which will be used to generate a value for a part or intermediary compose object's
+     * CustomTime Metadata Field. This will be a time set a duration in the future which will serve
+     * to aid in part cleanup via OLM Rules.
```

Comment:
Cleanup this comment. Items I think should be included:

1. Short description
2. link to the customTime property in our models
3. link to the olm rule docs for customTime
4. clarify the value will be computed each time a part or composed object is created and will be different for each part